wg-opcodes/summaries/20180214 - Meeting Summary.md

```diff
@@ -0,0 +1,108 @@
+ * Gavin Andresen
+ * Shammah Chancellor of Bitcoin ABC
+ * Andrew Stone of Bitcoin Unlimited
+# Discussion Topic
+
+To discuss the viability of OP_GROUP in regards to use case, safety, and the future of Bitcoin Cash.
+
+# Brief overview of discussion
+
+ * Rundown of the process to mint coloured coins
+     * Create a new address and send it some BCH with a normal P2SH or P2PKH transaction (This is your group address and a valid Bitcoin Cash address).  At that point you can send the BCH to any address and annotate that address with your group address. 
+     * Any unspent transaction output could be used to mint coloured coins. You do require a wallet that understands OP_GROUP to spend the UTXO
+     * The owner of the tokens retains control over the complete supply and can mint as required or they see fit.
+     * Proposal as it stands does not include a way to prevent the owner of a token “printing” more tokens.
+         * Chris Pacia has a proposal on how to do this using OP_CHAINHEIGHT (https://www.yours.org/content/colored-coins-in-bitcoin-cash-b26804e05964/ ) 
+ * Discussion on the possibility of putting out “half transactions” on the blockchain as a means to auction tokens on the blockchain, ( in relation to OP_GROUP use cases ).
+     * Questions regarding how the chosen “filler” of the transaction is decided.
+     * As the vendor of the “half transaction” you would wait and chose the most appealing “fill”
+     * The vendor signs the output, and waits for the input to be signed.
+     * Miners cant take the transaction because the OP_GROUP would not be balanced
+     * Using SIGHASH_SINGLE you have to ensure you don’t need a change address.
+     * Not an important use case for adoption, just an interesting and anonymous “auction” method
+
+ * Discussion on how OP_GROUP changes the way transactions are interpreted. 
+     * Inside the interpreter the OP_GROUP op code is a NO_OP.
+     * Clashes with the idea that script should be script and data is data
+ * Discussion on the Bitcoin Cash philosophy
+     * Thoughts on delaying features in favor of future code reorganization for the purpose of driving adoption  
+     * You have to support old transaction formats forever.
+ * Should OP_GROUP be a new transaction version?
+     * Requires more changes which need to be sold with the implementation
+     * Requires wallet support
+         * Cannot be tested without a wallet - BU has a full node + wallet available.
+             * https://github.com/gandrewstone/BitcoinUnlimited/tree/opgroup_fullnode
+             * https://github.com/gandrewstone/BitcoinUnlimited/tree/opgroup_consensus 
+         * 3 Months till Fork Date and 2 months to be fully tested
+     * Discussion about other aspects around changing the transaction version
+         * Is it limited to OP_GROUP
+         * Does it extend to other bugs than could be done with a TX version
+         * Do we bump the TX version with the reenabling of old OP_CODES
```

Comment:
I should think not.
 - The transaction format has not changed
 - Re-enabling is a hard fork so wallets have to upgrade anyway